src/org/linphone/setup/SetupActivity.java

```diff
@@ -69,17 +73,20 @@
 	final String registrarSRVLookupFormatTLS="_sips._tcp.%domain%";
 	//URL format for autoConfig lookup
 	final String autoConfigSRVLookupFormat="_rueconfig._tls.%domain%";
+	private Handler mHandler;
+	private ServiceWaitThread mThread;
 
 	static int WIFI_ACTIVITY_RESULT=0;
-
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 
 //		if (getResources().getBoolean(R.bool.isTablet) && getRequestedOrientation() != ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE) {
 //        	setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
 //        }
+		mHandler = new Handler();
 		
 		setContentView(R.layout.setup);
+		CDNProviders.getInstance().setContext(getApplicationContext());
 		firstFragment = getResources().getBoolean(R.bool.setup_use_linphone_as_first_fragment) ?
```

Comment:
Also ensure that you are calling setContext on CDNProviders in any other activity that may or may not touch SharedPreferences.